src/components/Accordionater.js

```diff
@@ -0,0 +1,86 @@
+import React, { Component } from 'react';
+import { AccordionBuilder, AccordionaterHeading } from './index';
+
+class Accordionater extends Component {
+	constructor(props) {
+		super(props);
+
+		this.state = {
+			panelCount: 1,
+		};
+
+		this.generateAccordionBuilders = this.generateAccordionBuilders.bind(this);
+		this.handleWizzyChange = this.handleWizzyChange.bind(this);
+		this.handleGenerateClodeClick = this.handleGenerateClodeClick.bind(this);
+		this.handleAddPanelClick = this.handleAddPanelClick.bind(this);
+		this.handleAccordionTitleChange = this.handleAccordionTitleChange.bind(this);
+		this.handleHeadingTypeChange = this.handleHeadingTypeChange.bind(this);
+	}
+
+	generateAccordionBuilders() {
+		let accordions;
+
+		for (let x = 0; x < this.state.panelCount; x += 1) {
+			accordions = accordions + (
+				<AccordionBuilder key={`panel${x}`} />
+			);
+		}
+
+		return accordions;
+	}
+
+	handleWizzyChange(title, value, index) {
+		if (this.props.onWizzyChange) {
+			this.props.onWizzyChange(title, value, index);
+		}
+	}
+
+	handleGenerateClodeClick() {
+		if (this.props.onGenerateCodeClick) {
+			this.props.onGenerateCodeClick();
+		}
+	}
+
+	handleAddPanelClick() {
+		this.setState({ panelCount: this.state.panelCount + 1})
+	}
+
+	handleAccordionTitleChange(value) {
+		if (this.props.onAccordionTitleChange) {
+			this.props.onAccordionTitleChange(value);
+		}
+	}
+
+	handleHeadingTypeChange(value) {
+		if (this.props.onHeadingTypeChange) {
+			this.props.onHeadingTypeChange(value);
+		}
+	}
+
+	render() {
+		const accordions = [];
+
+		for (let x = 0; x < this.state.panelCount; x += 1) {
```

Comment:
think you can use map here

```javascript
const accordions = this.state.panelCount.map((panelCount, index) => ... //do stuff here
```

Reply:
Nope - panelCount is an integer.

Reply:
doh -my bad

Reply:
I can do something like `Array(panelCount).fill().map((_, i) => i * i);` but it's more costly than a `for` loop, which are pretty cheap from a processing standpoint.

Reply:
yea im totally ok with this, i misread...this is not the comment you are looking for